### ISSUE TITLE
Make the moderation features optional

### ISSUE THREAD

**Issue by MrCordeiro** [Radi85/Comment#99]
I'm building a project in a context where moderation or flagging isn't needed. However, because of how the templatetags were done, I'm getting a lot of repeated or similar DB hits when rendering the comment list.

For what I could surmise, there is an extra hit for whenever the function `is_comment_moderator(user)` is called.

Maybe we could simplify all those searches with on/off switch in the product settings?

Another option would be to make some ORM optimization.

**Comment by abhiabhi94** [comment]
If I remember correctly, the flag option is disabled by default.

> Another option would be to make some ORM optimization.

Before going to the database, inside the function `is_comment_admin`,  we can check whether flags have been enabled or not. I think this should be good enough to satisfy our use-case.

If you have the time, it would be nice to accept new contributions to the code.

**Comment by MrCordeiro** [comment]
I think you are talking about these functions, right?

```python
def is_comment_admin(user):
    return user.groups.filter(name='comment_admin').exists() or \
           (user.has_perm('comment.delete_flagged_comment') and user.has_perm('comment.delete_comment'))

def is_comment_moderator(user):
    return user.groups.filter(name='comment_moderator').exists() or user.has_perm('comment.delete_flagged_comment')
```

I think it's the ` user.groups.filter(name='xxxx').exists()` the reason why I'm seeing those extra hits. I just forked the repo. Maybe I can get a PR in time for supper 🤞

**Comment by Radi85** [comment]
You are right, these 2 methods are useful only when flagging system is enabled. So I would return false before hitting the db for both of them when the flagging system is disabled.

**Comment by abhiabhi94** [comment]
> You are right, these 2 methods are useful only when flagging system is enabled

`comment_admins` can delete a comment irrespective of the flagging system. 

I am just pasting the relevant portions of the codebase to show this.

https://github.com/radi85/comment/tree/master/comment/templatetags/comment_tags.py#L108
```python
def can_delete_comment(comment, user):
    return is_comment_admin(user) or (comment.is_flagged and is_comment_moderator(user))
```

https://github.com/Radi85/Comment/blob/master/comment/templates/comment/comments/comment_content.html#L27
```jinja
{% if comment.user == user %}
    | {% include 'comment/comments/edit_icon.html' %}
    {% include 'comment/comments/delete_icon.html' %}
{% elif can_delete %}
    | {% include 'comment/comments/delete_icon.html' %}
    ...
```

**Comment by abhiabhi94** [comment]
merged in #101
